packages/core/src/shared/errors.ts

```diff
@@ -568,6 +582,16 @@ function hasCode<T>(error: T): error is T & { code: string } {
     return typeof (error as { code?: unknown }).code === 'string'
 }
 
+/**
+ * Returns the identifier the given error.
+ * Depending on the implementation, the identifier may exist on a
+ * different property.
+ */
+export function getErrorId(error: Error): string {
+    // prioritize code over the name
+    return hasCode(error) ? error.code : error.name
```

Comment:
should we rename `hasCode` to `hasId()`? though it's misleading because `error.code` is the standard field name. so maybe `getErrorId` should be `getErrorCode` ?

Reply:
I was thinking that if we refer to `code` it specifically means the `code` field. So `getErrorCode` may be misleading.

`id` is meant to be the generalized name that is either `code` or `name`.

---

I've seen cases where the error has a unique `name`, but also a `code` that provides additional context. I'm wondering if both exist, that we want them joined together to provide as much context as possible.

eg

```
const err = new Error('blah')
err.name = 'MyName'
err.code = 'MyCode'

getErrorId(err) // 'MyName:MyCode'
```

Reply:
@justinmk3